lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,431 @@
+            if key not in resource1:
+                if resource2[key] is not None:
+                    # Inexistent key is equivalent to exist with value None
+                    self.module.log(self.MSG_DIFF_AT_KEY.format(key) + debug_resources)
+                    return False
+
+        return True
+
+    def compare_list(self, first_resource, second_resource):
+        """
+        Recursively compares lists contents equivalence, ignoring types and element orders.
+        Lists with same size are compared value by value after a sort,
+        each element is converted to str before the comparison.
+        :arg list first_resource: first list
+        :arg list second_resource: second list
+        :return: True when equal; False when different.
+        """
+
+        resource1 = first_resource
+        resource2 = second_resource
+
+        debug_resources = "resource1 = {0}, resource2 = {1}".format(resource1, resource2)
+
+        # The second list is null / empty  / False
+        if not resource2:
+            self.module.log("resource 2 is null. " + debug_resources)
+            return False
+
+        if len(resource1) != len(resource2):
+            self.module.log("resources have different length. " + debug_resources)
+            return False
+
+        resource1 = sorted(resource1, key=self._str_sorted)
+        resource2 = sorted(resource2, key=self._str_sorted)
+
+        for i, val in enumerate(resource1):
+            if isinstance(val, collections.Mapping):
+                # change comparison function to compare dictionaries
+                if not self.compare(val, resource2[i]):
+                    self.module.log("resources are different. " + debug_resources)
+                    return False
+            elif isinstance(val, list):
+                # recursive call
+                if not self.compare_list(val, resource2[i]):
+                    self.module.log("lists are different. " + debug_resources)
+                    return False
+            elif self._standardize_value(val) != self._standardize_value(resource2[i]):
+                self.module.log("values are different. " + debug_resources)
+                return False
+
+        # no differences found
+        return True
+
+    def _str_sorted(self, obj):
+        if isinstance(obj, collections.Mapping):
+            return json.dumps(obj, sort_keys=True)
+        else:
+            return str(obj)
+
+    def _standardize_value(self, value):
+        """
+        Convert value to string to enhance the comparison.
+
+        :arg value: Any object type.
+
+        :return: str: Converted value.
+        """
+        if isinstance(value, float) and value.is_integer():
+            # Workaround to avoid erroneous comparison between int and float
+            # Removes zero from integer floats
+            value = int(value)
+
+        return str(value)
+
+    def merge_list_by_key(self, original_list, updated_list, key, ignore_when_null=[]):
```

Comment:
Looks like merge_list_by_key(), _standardize_value(), and _str_sorted() should be moved to be toplevel functions?  (They don't make use of self).